### ISSUE TITLE
Special dependency setup requirements.

### ISSUE THREAD

**Issue by BtbN** [airspeed-velocity/asv#650]
I'm having some issues with some of our dependencies.
Namely GDAL and rasterio.

About GDAL:
GDAL has to be installed in a version matching the binaries of the system.
So the usual pip invocation looks like:
`pip install --upgrade "gdal==$(gdal-config --version)" --install-option="build_ext" --install-option="$(gdal-config --cflags | sed 's/-I/--include-dirs=/')"`

The usual solution here is to install it manually before installing our package.

About rasterio:
rasterio is a bit odd, as setup.py itself has a dependency on numpy. So numpy has to be installed individually before even trying to install rasterio.

Is there a way to still achive this? Maybe provide an external plugin somehow, that installs those odd dependencies?

**Comment by pv** [comment]
If you need it now, it's possible to use a custom plugins like here:
https://github.com/OGGM/oggm/blob/master/asv_oggm_plugin.py
https://github.com/OGGM/oggm/blob/master/asv.conf.json

**Comment by BtbN** [comment]
Yep, that's in fact written by me on my work account. I just forgot I had opened this issue.
It's working fine like that against asv master for us.

I'll leave it to you if wehter to keep this issue open for future enhancements or to just close it as solved.
